module/VuFindSearch/src/VuFindSearch/Backend/EDS/Base.php

```diff
@@ -269,11 +269,14 @@ protected function parseAutocomplete($msg)
      */
     public function autocomplete($query, $type, $data, $raw = false)
     {
-        // build request
-        $url = $data['url'] . '?idx=' . urlencode($type) .
-            '&token=' . urlencode($data['token']) .
-            '&filters=[{"name"%3A"custid"%2C"values"%3A["' .
-            urlencode($data['custid']) . '"]}]&term=' . urlencode($query);
+        $params = [];
+        $params['idx'] = $type;
+        $params['token'] = $data['token'];
+        $params['filters'] = '[{"name" : "custid", "values": ["'.$data['custid'].'"]}]';
+        $params['term'] = $query;
+        
+        $url = $data['url'].'?'.http_build_query($params);
```

Comment:
I suspect that this is going to cause the build to fail on style grounds. I'd recommend running `vendor/bin/phing php-cs-fixer` to correct style issues. You can also `composer run qa` to make sure all of the checks will pass before pushing things up. I'm happy to help resolve any issues if something fails in an unclear way.

Reply:
Regarding the filter construction, this might be a slightly more concise approach to building the data structure:

<pre>
    $filters = [['name' => 'custid', 'values' => [$data['custid']]];
</pre>

Regarding `composer run qa`, is it possible you're working with an older version of VuFind? It was added relatively recently, so perhaps you just haven't pulled the latest dev code yet. It should be defined here:

https://github.com/vufind-org/vufind/blob/dev/composer.json#L129

Reply:
I am so sorry for making your life this hard for you  and thank you for humouring my beginner's mistakes.

I have adapted your suggested filter construction (I always get lost with all those square brackets, this one needed four closing ones 🤯, but it is definitely a lot nicer than my quick one.

I also added a unit test that will correctly assert the changed structure.

I ended up checking out the latest dev branch and got through most of running the composer run qa - it helped me find some more stuff, though it did fail... I might spend some more time on it on Monday

Reply:
@cwolfebsco, no need to apologize -- I'm really more than happy to help, and appreciate your willingness to respond to my feedback! Always happy to invest a little time in on-boarding, because it usually pays off in the long term.

I should note that I'll be on vacation for most of next week (out of office from 2/15-2/21), so I may be less responsive than usual during that period... but I'll be happy to continue to answer questions as they arise, though I may not be in a position to finalize this before I get back.

Reply:
...and look at that: the build just passed! I think this is quite possibly in a position to merge, but please let me know if you have more work you want to do before that happens.

Reply:
I hope you enjoy your vacation! I did run `composer run qa` again this morning and it exists after a timeout, but this seems to be more an issue of the resource constraint VM I am using. If you are happy with this one, I'd be happy to see it merged :)

Reply:
Thanks! I made a few last-minute tweaks (it looks like some of the style changes applied to the comments weren't necessary based on VuFind project standards, so I reverted them to keep the commit simple, and I reformatted the array construction a little more based on personal preferences, though that's more a matter of taste than anything). Merging now!